DocsCode/NavigationTutorial/NavigationTutorialViewController.swift

```diff
@@ -0,0 +1,121 @@
+        // #-end-code-snippet: navigation user-location-swift
+        
+        // #-code-snippet: navigation gesture-recognizer-swift
+        // Add a gesture recognizer to the map view
+        let setDestination = UILongPressGestureRecognizer(target: self, action: #selector(didLongPress(_:)))
+        mapView.addGestureRecognizer(setDestination)
+        // #-end-code-snippet: navigation gesture-recognizer-swift
+    }
+    
+    // #-code-snippet: navigation allow-callouts-swift
+    // Implement the delegate method that allows annotations to show callouts when tapped
+    func mapView(_ mapView: MGLMapView, annotationCanShowCallout annotation: MGLAnnotation) -> Bool {
+        return true
+    }
+    // #-end-code-snippet: navigation allow-callouts-swift
+    
+    // #-code-snippet: navigation long-press-swift
+    @objc func didLongPress(_ sender: UILongPressGestureRecognizer) {
+        guard sender.state == .began else { return }
+        
+        // Converts point where user did a long press to map coordinates
+        let point = sender.location(in: mapView)
+        let coordinate = mapView.convert(point, toCoordinateFrom: mapView)
+        
+        // Create a basic point annotation and add it to the map
+        let annotation = MGLPointAnnotation()
+        annotation.coordinate = coordinate
+        annotation.title = "Start navigation"
+        mapView.addAnnotation(annotation)
+        
+        // Calcuate the route from the user's location to the set destination
+        calculateRoute(from: (mapView.userLocation!.coordinate), to: annotation.coordinate) { (route, error) in
+            if error != nil {
+                NSLog("Error calculating route: \(String(describing: error))")
+            }
+        }
+    }
+    // #-end-code-snippet: navigation long-press-swift
+    
+    // #-code-snippet: navigation calculate-route-swift
+    // Calculate route to be used for navigation
+    func calculateRoute(from origin: CLLocationCoordinate2D,
+                        to destination: CLLocationCoordinate2D,
+                        completion: @escaping (Route?, Error?) -> ()) {
+        
+        // Coordinate accuracy is the maximum distance away from the waypoint that the route may still be considered viable, measured in meters. Negative values indicate that a indefinite number of meters away from the route and still be considered viable.
+        let origin = Waypoint(coordinate: origin, coordinateAccuracy: -1, name: "Start")
+        let destination = Waypoint(coordinate: destination, coordinateAccuracy: -1, name: "Finish")
+        
+        // Specify that the route is intented for automobiles avoiding traffic
+        let options = NavigationRouteOptions(waypoints: [origin, destination], profileIdentifier: .automobileAvoidingTraffic)
+        
+        // Generate the route object and draw it on the map
+        Directions.shared.calculate(options) { (waypoints, routes, error) in
```

Comment:
We need `_ =` to avoid a warning about an unused return value. We can remove it when https://github.com/mapbox/MapboxDirections.swift/pull/209 makes it into a release.